Lib/site.py

```diff
@@ -390,6 +390,23 @@ def setcopyright():
         files, dirs)
 
 
+def _register_detect_pip_usage_in_repl():
+    old_excepthook = sys.excepthook
+
+    def detect_pip_usage_in_repl(typ, value, traceback):
+        if typ is SyntaxError and (
+            "pip install" in value.text or "pip3 install" in value.text
+        ):
+            value.msg = ("The Python package manager (pip) can only be used"
+                         " from outside of Python.\n"
+                         "Please try the `pip` command in a"
+                         " separate terminal or command prompt.")
+
+        old_excepthook(typ, value, traceback)
+
+    sys.excepthook = detect_pip_usage_in_repl
```

Comment:
Also set `detect_pip_usage_in_repl.__wrapped__ = old_excepthook`

Reply:
Ok, I've done this.